tools/spec-gen-sdk/src/utils/readme.ts

```diff
@@ -0,0 +1,81 @@
+import * as commonmark from "commonmark";
+import * as jsYaml from "js-yaml";
+
+/**
+ * The parsed version of the swagger-to-sdk YAML block within an AutoRest readme.md file.
+ */
+export interface ReadmeMdSwaggerToSDKConfiguration {
+  /**
+   * The repositories specified.
+   */
+  repositories: RepositoryConfiguration[];
+}
+
+/**
+ * An individual repository configuration within an AutoRest readme.md swagger-to-sdk YAML block
+ * configuration.
+ */
+export interface RepositoryConfiguration {
+  /**
+   * The name of the GitHub repository this configuration applies to. If no organization is
+   * specified, then Azure will be used.
+   */
+  repo: string;
+  /**
+   * The list of commands that will be run (in order) after an SDK has been generated.
+   */
+  after_scripts: string[];
+}
+
+function findSwaggerToSDKYamlBlocks(parsedMarkdownNode: commonmark.Node | undefined | null): commonmark.Node[] {
+  const result: commonmark.Node[] = [];
+  if (parsedMarkdownNode) {
+    const nodesToVisit: commonmark.Node[] = [parsedMarkdownNode];
+    while (nodesToVisit.length > 0) {
+      const node: commonmark.Node = nodesToVisit.shift()!;
+
+      if (node.firstChild) {
+        nodesToVisit.push(node.firstChild);
+      }
+      if (node.next) {
+        nodesToVisit.push(node.next);
+      }
+
+      if (node.type === "code_block" && node.info && node.info.toLowerCase().indexOf("$(swagger-to-sdk)") !== -1) {
+        result.push(node);
+      }
+    }
+  }
+  return result;
+}
+
+/**
+ * Parse the contents of an AutoRest readme.md configuration file and return the parsed swagger to
+ * sdk configuration section.
+ * @param readmeMdFileContents The contents of an AutoRest readme.md configuration file.
+ */
+export function findSwaggerToSDKConfiguration(readmeMdFileContents: string | undefined): ReadmeMdSwaggerToSDKConfiguration | undefined {
```

Comment:
is this something that could be used by other jobs in the specs repo that would make more sense to have in there than in here?

Reply:
It might be used in other place where requires to access the SDK configuration in `readme.md`. But we don't have an individual tool staying in spec repo and is published as a library in order to be consumed.

Reply:
Does the flow before this involves cloning the specs repo? What if there is a common script on the spec repo that different tools can call? no need to publish anything

Reply:
The new design won't have dependency on the spec repo. It makes sense to keep this function in the same code base or reference it from another npm library.